CMake/Findzstd.cmake

```
@@ -50,3 +50,13 @@ if(NOT TARGET zstd::zstd)
     zstd::zstd PROPERTIES IMPORTED_LINK_INTERFACE_LANGUAGES "C"
                           IMPORTED_LOCATION "${ZSTD_LIBRARIES}")
 endif()
+
+# Create canonical alias targets regardless of how zstd::zstd was provided.
+if(TARGET zstd::zstd)
+  if(NOT TARGET zstd::libzstd_shared)
+    add_library(zstd::libzstd_shared ALIAS zstd::zstd)
+  endif()
+  if(NOT TARGET zstd::libzstd_static)
+    add_library(zstd::libzstd_static ALIAS zstd::zstd)
+  endif()
+endif()
```

This is what I meant by not including the `if(TARGET zstd::zstd)`:

```suggestion
if(NOT TARGET zstd::libzstd_shared)
  add_library(zstd::libzstd_shared ALIAS zstd::zstd)
endif()

if(NOT TARGET zstd::libzstd_static)
  add_library(zstd::libzstd_static ALIAS zstd::zstd)
endif()
```

Or do we need it for some reason?

---

no, you are right